File: lista0/exercicio2_nadador/main.cpp
#include <iostream>
#include <vector>
#include <limits>
#include <string>

struct RegraCategoria {
    int idadeLimite;
    std::string nome;
};

int main() {
    int idade{0};

    std::vector<RegraCategoria> tabela = {
        {4,  "Muito jovem para competir"},
        {7,  "Infantil A"},
        {10, "Infantil B"},
        {13, "Juvenil A"},
        {17, "Juvenil B"}
    };
    std::string categoriaSenior = "Senior";

    std::cout << "Digite a idade: ";
    while(!(std::cin >> idade)){
        std::cin.clear();
        std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
        std::cout << "Invalido. Digite a idade: ";
    }

    std::string resultado = categoriaSenior; 
    for (const auto& regra : tabela) {
        if (idade <= regra.idadeLimite) {
            resultado = regra.nome;
            break;
        }
    }

    std::cout << "Categoria: " << resultado << std::endl;

    return 0;
}

File: lista0/exercicio2_nadador/main.rs
use std::io;

fn main(){
    let mut idade = String::new();
    io::stdin().read_line(&mut idade).expect("Idade invalida!");
    let idade: i32 = idade.trim().parse().expect("Erro ao converter idade");
    
    if idade <= 4 {
        println!("Muito jovem pra competir");
    } else if idade <= 7{
        println!("Infantil A");
    }
    else if idade <= 10{
        println!("Infantil B");
    }
    else if idade <= 13{
        println!("Juvenil A");
    }
    else if idade <= 17{
        println!("Juvenil B");
    }
    else{
        println!("Senior")
    }

}
